src/tests/ftest/server/daos_server_restart.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from apricot import TestWithServers
+from server_utils import ServerFailed
+from daos_utils import DaosCommand
+from command_utils import CommandFailure
+
+
+class DaosServerTest(TestWithServers):
+    """Daos server tests.
+
+    Test Class Description:
+        Test to verify that the daos_io_server, daos_server starts/stops
+        and reformat.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosServerTest object."""
+        super(DaosServerTest, self).__init__(*args, **kwargs)
+
+    def restart_daos_server(self, reformat=True):
```

Comment:
This function seems to be duplicating what is being tested in `dmg_storage_reformat.py` test. Instead of having this as a new test, I would create it as a new test case within the `dmg_storage_reformat.py` file if needed. I think instead, the `dmg system stop` and `dmg system strart` commands should be used and reformat the system in between like Tom mentioned in his comments below.

Reply:
This test includes multiple server restarts, with pool and container creation, and verify reformat as a clean restart. 
steps:
        self.log.info("(1)Verify daos server pool list after started.")
        self.log.info("(2)Restart server without pool created and verify.")
        self.log.info("(3)Create pools, containers.")
        self.log.info("(4)Shutdown, restart and reformat the server")
        self.log.info("(5)Verify after server restarted.")

Reply:
Yes, I understand the steps. What I'm pointing out is that although the `dmg_storage_reformat.py` test is implemented in a different way, they seem to be testing the same functionality.